src/notification_service/sender/base.py

```diff
@@ -0,0 +1,21 @@
+from abc import ABC, abstractmethod
```

Comment:
Wouldn't it be better to name class and file with the same name? What do you think?

Reply:
I called it base just to emphasize that this class is a base class. But I can rename it as well.

Reply:
I've seen from some best practises that if a module defines a single class, the module name and class name should match. It increases readability when you want to find a code of the class.
Like your example:
class name = SenderProvider
module_name = sender_provider.py

If you do not mind, let's use consistent naming throughout the code where we have one module = one class :)

Reply:
Thank you for the research, I didn't know that. I will change the name then 👍